// Copyright 2015 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

import "compositor_message_to_impl.proto";
import "compositor_message_to_main.proto";
import "layer_tree_host.proto";

package cc.proto;

// A top level message for serialized state used by the compositor remote
// channels.
// TODO(khushalsagar): Remove CompositorMessage and stop using
// RemoteProtoChannel to plumb these to/from the browser. See crbug.com/648442
message CompositorMessage {
  // Engine -> Client
  // Frame Update.
  optional LayerTreeHost layer_tree_host = 3;

  // One of these is set based on where the message is going to.
  optional CompositorMessageToMain to_main = 1;
  optional CompositorMessageToImpl to_impl = 2;
}
